### PR TITLE
Bugfixes for EOF, whitespace, and prettier checks.

### PR DESCRIPTION
**Pull request by pboulos**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [x] If there is a related issue, make sure it is linked to this PR.

**Description of changes**

When running CI, we observe the following errors due to linting.

```
🚀 Checking Poetry lock file consistency with 'pyproject.toml': Running poetry lock --check
All set!
🚀 Linting code: Running pre-commit
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/astral-sh/ruff-pre-commit.
[INFO] Initializing environment for https://github.com/pre-commit/mirrors-prettier.
[INFO] Initializing environment for https://github.com/pre-commit/mirrors-prettier:prettier@3.0.3.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/astral-sh/ruff-pre-commit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/mirrors-prettier.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
check for case conflicts.................................................Passed
check for merge conflicts................................................Passed
check toml...............................................................Passed
check yaml...............................................................Passed
fix end of files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing LICENSE

trim trailing whitespace.................................................Failed
- hook id: trailing-whitespace
- exit code: 1
- files were modified by this hook

Fixing .github/workflows/on-release-main.yml

ruff.....................................................................Passed
ruff-format..............................................................Passed
prettier.................................................................Failed
- hook id: prettier
- files were modified by this hook

README.md
.github/actions/setup-poetry-env/action.yml
.github/workflows/main.yml
.github/workflows/on-release-main.yml
mkdocs.yml

make: *** [Makefile:13: check] Error 1
Error: Process completed with exit code 2.
```

This PR applies some formatting changes across the repo such that these tests are now passed in the initial commit. This appears to be related to https://github.com/fpgmaas/cookiecutter-poetry/issues/96 and https://github.com/fpgmaas/cookiecutter-poetry/pull/97.

Changes include:
- Reformatting the extensions item in `{{cookiecutter.project_name}}/.devcontainer/devcontainer.json` to be on a single line.
- Un-indenting lines 5 through 8 in `{{cookiecutter.project_name}}/.github/actions/setup-poetry-env/action.yml`.
- Removing a newline on line 51 in `{{cookiecutter.project_name}}/.github/workflows/main.yml`.
- Jinja templating changes for lstrip and trim blocks in `{{cookiecutter.project_name}}/.github/workflows/on-release-main.yml`.
- Jinja templating changes for lstrip and trim blocks in `{{cookiecutter.project_name}}/README.md`.
- Un-indenting in `{{cookiecutter.project_name}}/mkdocs.yml`.

These changes should not change any user behaviour apart from allowing the CI to pass first time without any user modification.

### CODE REVIEW

**Comment by pboulos** [comment]
@fpgmaas I wanted to ask if you could take a look at this. They've been errors that I've been encountering while using this template that I've needed to manually fix before using it every time; I thought it would be useful to make the PR for others to benefit from.

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "cookiecutter-poetry"
-version = "0.1.0"
+version = "0.1.1"
```
Comment (fpgmaas) [inline pyproject.toml:3]:
```suggestion
version = "0.1.0"
```

This can be kept unchanged.

**Comment by fpgmaas** [comment]
> @fpgmaas I wanted to ask if you could take a look at this. They've been errors that I've been encountering while using this template that I've needed to manually fix before using it every time; I thought it would be useful to make the PR for others to benefit from.

hanks for the contribution, and sorry for the late response! I was on holiday the past three weeks. I left one small comment; I'm happy to merge the PR after that is processed.

File: pyproject.toml (hunk 1 above)
Comment (pboulos) [reply pyproject.toml]: Did this, thanks.

**Pull request merged**